crates/forge/src/test_execution_syscall_handler.rs

```diff
@@ -60,15 +60,34 @@ impl From<&StarknetContractArtifacts> for ContractArtifacts {
     }
 }
 
+pub struct TestExecutionState<'a> {
```

Comment:
I don't like the `State` in name, it suggests mutability and blockchain state. Both of the fields of this struct don't ever change during execution, those are more like `TestExecutionParams` (can use a better name though)

Reply:
This is state, which can be mutable if we want. It is for consistency with CheatnetState.